companion/src/shared/timeredit.cpp

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (C) OpenTX
+ *
+ * Based on code named
+ *   th9x - http://code.google.com/p/th9x
+ *   er9x - http://code.google.com/p/er9x
+ *   gruvin9x - http://code.google.com/p/gruvin9x
+ *
+ * License GPLv2: http://www.gnu.org/licenses/gpl-2.0.html
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "timeredit.h"
+#include <cmath>
+
+TimerEdit::TimerEdit(QWidget * parent) :
+  QLineEdit(parent),
+  m_showSeconds(true),
+  m_minTime(0),
+  m_maxTime(24 * 3600 - 1),
+  m_singleStep(1),
+  m_pageStep(60),
+  m_hourDigits(2),
+  m_validator(new QRegularExpressionValidator(this))
+{
+  setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Fixed);
+  setupFormat();
+  setToolTip(tr("Use mouse scroll or UP/DOWN arrow keys to change time in small steps.\nCTRL + scroll or PAGE UP/DOWN keys to change time in larger steps."));
+  connect(this, &TimerEdit::textEdited, this, &TimerEdit::textEditedHandler);
+}
+
+int TimerEdit::timeInSeconds() const
+{
+  int ret = 0;
+
+  ret += timepart(text(), TimeParts::Hours) * 3600;
+  ret += timepart(text(), TimeParts::Minutes) * 60;
+  ret += timepart(text(), TimeParts::Seconds);
+  ret *= timepart(text(), TimeParts::Polarity);
+
+  //qDebug() << ret;
+  return ret;
+}
+
+int TimerEdit::timepart(const QString & input, TimerEdit::TimeParts part) const
+{
+  int ret = 0;
+  QRegularExpressionMatch match = m_validator->regularExpression().match(input, 0, QRegularExpression::NormalMatch);
+
+  if (match.hasMatch()) {
+    switch (part) {
+      case Hours:
+        if (!match.captured("hrs").isEmpty())
+          ret = match.captured("hrs").toInt();
+        break;
+
+      case Minutes:
+        if (!match.captured("mins").isEmpty())
+          ret = match.captured("mins").toInt();
+        break;
+
+      case Seconds:
+        if (!match.captured("secs").isEmpty())
+          ret = match.captured("secs").toInt();
+        break;
+
+      case Polarity:
+        if (!match.captured("pol").isEmpty() && !match.captured("pol").compare("-"))
+          ret = -1;
+        else
+          ret = 1;
+        break;
+
+      default:
+        break;
+    }
+  }
+  //qDebug() << input << match << ret;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this block of commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3ACommentedCode)